### ISSUE TITLE
Make sure there are no hanging promises

### ISSUE THREAD

**Issue by IanKrieger** [brave/ads-ui#1042]
enable the below eslint config and fix any errors:
```
    "@typescript-eslint/no-floating-promises": "error",
    "@typescript-eslint/no-misused-promises": "error",
```

**Comment by shubrautt** [comment]
@IanKrieger can I work on this issue?

**Comment by IanKrieger** [comment]
Yes you can, @shubrautt

**Comment by shubrautt** [comment]
![image](https://github.com/brave/ads-ui/assets/60537610/e76e6c24-05de-42b9-bfa3-d9eda821e74a)
@IanKrieger should I add void keyword or need to handle the promise?

**Comment by IanKrieger** [comment]
I think in cases where we are setting a form value, we should `await` them, making the onClick async: `onClick{async () => ...}`

**Comment by shubrautt** [comment]
I think we need to set checksVoidReturn to be false, to avoid this error `@typescript-eslint/no-misused-promises": ["error", { "checksVoidReturn": false }]`

**Comment by IanKrieger** [comment]
I don't think we need that rule, but If you could provide an example of a line in the codebase that is failing because this check does not exist, we can go from there and I can give some direction on how to handle that.

**Comment by shubrautt** [comment]
This is the file where I'm getting error.
![image](https://github.com/brave/ads-ui/assets/60537610/a8abd2e9-a965-4766-9e64-7ab42d35491b)

**Comment by shubrautt** [comment]
Same error also exist on other files also

**Comment by IanKrieger** [comment]
Hmm, well since I do wish for the value updates to propagate, and I do not like the workarounds I've seen, I am fine with `{ "checksVoidReturn": false }`

Thank you for asking

**Comment by shubrautt** [comment]
![image](https://github.com/brave/ads-ui/assets/60537610/cc7f7cd8-2359-428c-b7a7-237d204503f2)

what should I use while solving no-floating-promises errors, async/await or then/catch?

**Comment by shubrautt** [comment]
#1049 I have pushed the code, how can I able to figure out where things need to be handled.
